collect/collect.go

```diff
@@ -620,18 +620,18 @@ func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span) (processed bool,
 // dealWithSentTrace handles a span that has arrived after the sampling decision
 // on the trace has already been made, and it obeys that decision by either
 // sending the span immediately or dropping it.
-func (i *InMemCollector) dealWithSentTrace(ctx context.Context, tr cache.TraceSentRecord, sentReason string, sp *types.Span) {
+func (i *InMemCollector) dealWithSentTrace(ctx context.Context, tr cache.TraceSentRecord, keptReason string, sp *types.Span) {
 	_, span := otelutil.StartSpanMulti(ctx, i.Tracer, "dealWithSentTrace", map[string]interface{}{
 		"trace_id":    sp.TraceID,
-		"sent_reason": sentReason,
+		"kept_reason": keptReason,
 		"hostname":    i.hostname,
 	})
 	defer span.End()
 
 	if i.Config.GetAddRuleReasonToTrace() {
 		var metaReason string
-		if len(sentReason) > 0 {
-			metaReason = fmt.Sprintf("%s - late arriving span", sentReason)
+		if len(keptReason) > 0 {
+			metaReason = fmt.Sprintf("%s - late arriving span", keptReason)
 		} else {
 			metaReason = "late arriving span"
 		}
```

Comment:
maybe "meta.refinery.send_reason" also needs a rename? But it will break people's metrics dashboards.

Reply:
The `meta.refinery.send_reason` is recording the reason why a trace is moved into the decision making process. We use `meta.refinery.reason` to record the `keptReason`. We should be ok to keep it as it is